src/main/java/org/jabref/gui/preferences/PreferencesDialog.java

```diff
@@ -127,6 +128,13 @@ private void construct() {
                                                 .withText(PrefsTab::getTabName)
                                                 .install(tabsList);
 
+        TextField searchBox = new TextField();
+        searchBox.setPromptText(Localization.lang("Search"));
+
+        PreferencesSearchHandler searchHandler = new PreferencesSearchHandler(preferenceTabs);
+        tabsList.itemsProperty().bindBidirectional(searchHandler.filteredPreferenceTabsProperty());
```

Comment:
I'm not sure if this is the reason for the problem, but it should suffice to replace the `filteredPreferenceTabsProperty` by an observable list (instead of an list property) and then use the `bindContentsBidirectional` method to bind `tabsList.getItems()` to this observable list.

Reply:
Unfortunately, this did not work either. The problem is that `searchHandler.searchText` does not reflect the changes from the `searchBox`, unless an explicit listener is added to `searchBox` and `searchHandler.searchText` is simply accessed there. 

```
searchBox.textProperty().addListener((o, p, n) -> {
    searchHandler.searchTextProperty();     // required, so that changed are reflected.
});
```